zcash_client_backend/src/data_api.rs

```diff
@@ -593,11 +595,115 @@ impl BlockMetadata {
 
     /// Returns the size of the Orchard note commitment tree for the final treestate of the block
     /// that this [`BlockMetadata`] describes, if available.
+    #[cfg(feature = "orchard")]
     pub fn orchard_tree_size(&self) -> Option<u32> {
         self.orchard_tree_size
     }
 }
 
+/// The Sapling note commitment and nullifier data extracted from a [`CompactBlock`] that is
+/// required by the wallet for note commitment tree maintenance and spend detection.
+///
+/// [`CompactBlock`]: crate::proto::compact_formats::CompactBlock
+pub struct ScannedBlockSapling {
```

Comment:
It probably makes sense for this instead to be a polymorphic `ScannedBundle` type to eliminate the duplication.